Client/Frontend/Browser/BrowserViewController/BrowserViewController+Menu.swift

```diff
@@ -0,0 +1,94 @@
+                let vc = HistoryViewController(isPrivateBrowsing: PrivateBrowsingManager.shared.isPrivateBrowsing)
+                vc.toolbarUrlActionsDelegate = self
+                menuController.pushViewController(vc, animated: true)
+            }
+            MenuItemButton(icon: #imageLiteral(resourceName: "menu-downloads").template, title: Strings.downloadsMenuItem) {
+                let vc = DownloadsPanel(profile: self.profile)
+                vc.applyTheme(Theme.of(nil))
+                menuController.pushViewController(vc, animated: true)
+            }
+            MenuItemButton(icon: #imageLiteral(resourceName: "menu-settings").template, title: Strings.settingsMenuItem) {
+                let vc = SettingsViewController(profile: self.profile, tabManager: self.tabManager, feedDataSource: self.feedDataSource, rewards: self.rewards, legacyWallet: self.legacyWallet)
+                vc.settingsDelegate = self
+                menuController.pushViewController(vc, animated: true)
+            }
+        }
+    }
+    
+    func activitiesMenuSection(_ menuController: MenuViewController, tabURL: URL, activities: [UIActivity]) -> some View {
+        VStack(alignment: .leading, spacing: 0) {
+            MenuTabDetailsView(tab: tabManager.selectedTab, url: tabURL)
+            VStack(spacing: 0) {
+                MenuItemButton(icon: #imageLiteral(resourceName: "menu-add-bookmark").template, title: Strings.addToMenuItem) {
+                    self.dismiss(animated: true) {
+                        self.openAddBookmark()
+                    }
+                }
+                ForEach(activities, id: \.activityTitle) { activity in
+                    MenuItemButton(icon: activity.activityImage?.template ?? UIImage(), title: activity.activityTitle ?? "") {
+                        self.dismiss(animated: true) {
+                            activity.perform()
+                        }
+                    }
+                }
+                MenuItemButton(icon: #imageLiteral(resourceName: "nav-share").template, title: Strings.shareWithMenuItem) {
+                    self.dismiss(animated: true)
+                    self.tabToolbarDidPressShare()
+                }
+            }
+        }
+    }
+    
+    struct MenuTabDetailsView: View {
+        @SwiftUI.Environment(\.colorScheme) var colorScheme: ColorScheme
+        var tab: Tab?
+        var url: URL
+        
+        var body: some View {
+            VStack(alignment: .leading, spacing: 2) {
+                if let tab = tab {
+                    Text(verbatim: tab.displayTitle)
+                        .font(.callout)
+                        .fontWeight(.medium)
+                        .lineLimit(1)
+                        .foregroundColor(Color(Theme.of(nil).colors.tints.home))
+                }
+                Text(verbatim: url.baseDomain ?? url.absoluteDisplayString)
```

Comment:
Maybe `url.baseDomain ?? url.host ?? url.absoluteDisplayString`? That way you attempt to get base domain else eTLD else display string?

Reply:
Good call, `absoluteDisplayString` only strips http/https so `host` is probably a better fallback